app/org/sagebionetworks/bridge/services/UploadSchemaService.java

```diff
@@ -0,0 +1,86 @@
+package org.sagebionetworks.bridge.services;
+
+import com.google.common.base.Strings;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.dao.UploadSchemaDao;
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.models.studies.Study;
+import org.sagebionetworks.bridge.models.upload.UploadSchema;
+import org.sagebionetworks.bridge.validators.Validate;
+
+/**
+ * Service handler for upload schema APIs. This is called by researchers to create, read, and update upload schemas.
+ */
+@Component
+public class UploadSchemaService {
+    private UploadSchemaDao uploadSchemaDao;
+
+    /** DAO for upload schemas. This is configured by Spring. */
+    @Autowired
+    public void setUploadSchemaDao(UploadSchemaDao uploadSchemaDao) {
+        this.uploadSchemaDao = uploadSchemaDao;
+    }
+
+    /**
+     * <p>
+     * Service handler for creating and updating upload schemas. This method creates an upload schema, using the study
+     * ID and schema ID of the specified schema, or updates an existing one if it already exists.
+     * </p>
+     * <p>
+     * Preconditions: The study (provided and validated by the controller) is compared against the schema to ensure
+     * that the schema is being created under the correct study.
+     * </p>
+     * <p>
+     * This method validates the schema. However, it does not validate the study, as that is not user input.
+     * </p>
+     *
+     * @param study
+     *         the study for this endpoint, provided by the controller, used for validation
+     * @param uploadSchema
+     *         schema to create or update, must be non-null, must contain a valid study ID and a valid schema ID
+     * @return the created or updated schema, will be non-null
+     */
+    public UploadSchema createOrUpdateUploadSchema(Study study, UploadSchema uploadSchema) {
+        // validate schema
+        if (uploadSchema == null) {
+            throw new InvalidEntityException(String.format(Validate.CANNOT_BE_NULL, "upload schema"));
+        }
+        Validate.entityThrowingException(UploadSchema.Validator.INSTANCE, uploadSchema);
+
+        // validate that the schema's study ID matches the server endpoint's study
+        if (!uploadSchema.getStudyId().equals(study.getIdentifier())) {
```

Comment:
Discussed this with Eric. I'll add a @JsonIgnore to the study property so that it's never exposed to the caller, and we'll just use the endpoint-derived study.